core-bundle/src/Resources/views/Frontend/preview_toolbar_base_js.html.twig

```diff
@@ -233,25 +233,36 @@
                 return;
             }
 
+            const userField = form.querySelector('input[name="user"]');
+
             form.addEventListener('submit', function (e) {
                 e.preventDefault();
                 const formData = new FormData(form);
 
                 request('POST', form.action, formData, function () {
-                    if (204 === this.status) {
-                        location.reload();
+                    if (400 === this.status) {
```

Comment:
This should be 401, too.

Reply:
See above. The response will be `401` if you are not authenticated (in which case we want to reload the preview page so that you can log in again) and `400` if the provided username does not exist.